mypyc/lib-rt/getargs.c

```diff
@@ -1269,14 +1279,30 @@ vgetargskeywords(PyObject *args, PyObject *kwargs, const char *format,
                                  " (%zd given)",
                                  (fname == NULL) ? "function" : fname,
                                  (fname == NULL) ? "" : "()",
-                                 (min != INT_MAX) ? "at most" : "exactly",
+                                 (min < max) ? "at most" : "exactly",
                                  max,
                                  max == 1 ? "" : "s",
                                  nargs);
                 }
                 return cleanreturn(0, &freelist);
             }
         }
+        if (*format == '@') {
+            if (min == INT_MAX && max == INT_MAX) {
+                PyErr_SetString(PyExc_SystemError,
+                                "Invalid format string "
+                                "(@ without preceding | and $)");
```

Comment:
I added the error checks here to stay consistent with CPython (and because I'm trying to get a version of it merged into CPython), but my longer term plan is to strip most of these checks (well, or to turn them into asserts) as we further customize getargs.